ql/src/java/org/apache/hadoop/hive/ql/parse/MergeSemanticAnalyzer.java

```diff
@@ -142,31 +125,23 @@ WHEN NOT MATCHED THEN INSERT VALUES (source.a2, source.b2)
     }
     List<ASTNode> whenClauses = findWhenClauses(tree, whenClauseBegins);
 
-    StringBuilder rewrittenQueryStr = createRewrittenQueryStrBuilder();
-    rewrittenQueryStr.append("(SELECT ");
     String subQueryAlias = isAliased(targetNameNode) ? targetName : targetTable.getTTable().getTableName();
-    ColumnAppender columnAppender = getColumnAppender(subQueryAlias, StringUtils.EMPTY);
-    columnAppender.appendAcidSelectColumns(rewrittenQueryStr, Context.Operation.MERGE);
-
-    rewrittenQueryStr.deleteCharAt(rewrittenQueryStr.length() - 1); // remove last ','
-    addColsToSelect(targetTable.getCols(), rewrittenQueryStr);
-    addColsToSelect(targetTable.getPartCols(), rewrittenQueryStr);
-    rewrittenQueryStr.append(" FROM ").append(getFullTableNameForSQL(targetNameNode)).append(") ");
-    rewrittenQueryStr.append(subQueryAlias);
-    rewrittenQueryStr.append('\n');
+    MergeRewriter mergeRewriter = new MergeRewriterFactory(conf).createMergeRewriter(
```

Comment:
should we create a new factory instance every time analyze is called or init it in constructor where we set the conf obj?

Reply:
Moved factory creation to SemanticAnalyzerFactory